src/point.h

```diff
@@ -380,4 +393,63 @@ inline const std::array<tripoint, 8> eight_horizontal_neighbors = { {
     }
 };
 
+/* Return points in a clockwise spiral from min_dist to max_dist, inclusive, ordered by
+ * the closest points first. The distance is calculated using roguelike distances, so
+ * movement in any direction only counts as 1.
+ *
+ * The predicate function is evaluated on each point. If the function returns true, the
+ * point is returned. If the predicate function never returns true then std::nullopt is
+ * returned once max_dist is reached.
+ *
+ * @param center The center of the spiral.
+ * @param min_dist minimum distance to start the spiral from.
+ * @param max_dist greatest distance from the centre that the spiral will go to.
+ * @returns std::nullopt if min_dist > max_dist or predicate_fn evaluated to true for
+ *      no points.
+ */
+template <typename PredicateFn, typename Point>
+std::optional<Point> find_point_closest_first( const Point &center, int min_dist, int max_dist,
+        PredicateFn &&predicate_fn )
+{
+    std::optional<int> n = rectangle_size( min_dist, max_dist );
+
+    if( n == std::nullopt ) {
+        return {};
+    }
+
+    const bool is_center_included = min_dist == 0;
+
+    if( is_center_included ) {
+        if( predicate_fn( center ) ) {
+            return center;
+        }
+    }
```

Comment:
Do we really need the explicit declaration of this bool that won't be used anywhere else except for the next check?
```suggestion
    if( min_dist == 0 && predicate_fn( center ) ) {
        return center;
    }
```